packages/react/src/components/HorizontalRule/HorizontalRule.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+import settings from 'carbon-components/es/globals/js/settings';
+
+import '@ibmdotcom/styles/scss/components/horizontalrule/_horizontalrule.scss';
+
+const { prefix } = settings;
+/**
+ * Horizontal Rule component
+ *
+ * @param {object} props props object
+ * @param {string} props.type type of rule (solid or dashed)
+ * @param {string} props.color color of the rule (low-contrast, medium-contrast, or high-contrast)
+ * @param {string} props.width width of the rule (thin or thick)
+ * @param {string} props.length length of rule (small, medium, large, inset, or overhung)
+ * @returns {*} Horizontal Rule component
+ */
+const HorizontalRule = ({ type, length, color, width }) => (
+  <hr
+    data-autoid="hr"
+    className={classnames(
+      `${prefix}--hr`,
+      `${prefix}--hr__${type}__${color}`,
+      { [`${prefix}--hr__dashed`]: type === 'dashed' },
+      `${prefix}--hr__${length}`,
+      `${prefix}--hr__${width}`
+    )}
+  />
+);
+
+HorizontalRule.propTypes = {
+  type: PropTypes.string,
+  length: PropTypes.string,
+  color: PropTypes.string,
+  width: PropTypes.string,
+};
+
+HorizontalRule.defaultProps = {
```

Comment:
saw your comment on using function to clean up the modifiers
```
function hrMod (mod) {
  return `${prefix}--hr--${mod}`;
}
```

we could check for the prop value before setting up the classname

however I do like having the default values in the JS so we can quickly see what the defaults are instead of going through the `CSS`

Reply:
@annawen1 I think it would be great to maintain our separation of concerns here. I don't think it would bother me as much if we were only building these out in React, but how this will play out with vanilla is the user will need a number of classes to just get a simple `<hr />` to show up.

Reply:
got it - removed the defaultProps and set the default in the CSS